File: an2_sem4/MPP/Homework/sapt14/network/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/network/server/ConcurrentServer.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.server;

import java.net.Socket;

public abstract class ConcurrentServer extends AbstractServer {
    public ConcurrentServer(int port) {
        super(port);
    }

    @Override
    protected void processRequest(Socket client) {
        Thread tw=createWorker(client);
        tw.start();
    }

    protected abstract Thread createWorker(Socket client);
}

File: an2_sem4/MPP/Homework/sapt11/cs/FestivalSellpoint/FestivalSellpoint.Network/Server/ConcurrentServer.cs
using System.Net.Sockets;
using System.Threading;

namespace FestivalSellpoint.Network.Server
{
    public abstract class ConcurrentServer : AbstractServer
    {
        public ConcurrentServer(string host, int port) : base(host, port) { }

        public override void ProcessRequest(TcpClient client)
        {
            CreateWorker(client).Start();
        }

        protected abstract Thread CreateWorker(TcpClient client);
    }
}
